src/Microsoft.AspNet.Mvc.Razor.Host/MvcRazorCodeParser.cs

```diff
@@ -105,14 +112,14 @@ protected virtual void InjectDirective()
             // typeName now contains the token 'MyApp.MyService'
             var typeName = Span.GetContent().Value;
 
-            var propertyStartLocation = CurrentLocation;
             AcceptWhile(IsSpacingToken(includeNewLines: false, includeComments: true));
 
             if (!hasTypeError && (EndOfFile || At(CSharpSymbolType.NewLine)))
             {
                 // Add an error for the property name only if we successfully read the type name
-                Context.OnError(propertyStartLocation,
-                                Resources.FormatMvcRazorCodeParser_InjectDirectivePropertyNameRequired(InjectKeyword));
+                Context.OnError(startLocation,
+                                Resources.FormatMvcRazorCodeParser_InjectDirectivePropertyNameRequired(InjectKeyword),
+                                keywordwithSingleWhitespaceLength + remainingWhitespaceLength);
```

Comment:
This is going to look strange.  Suggest the error include all of the directive that was present -- the type name as well as the keyword?